lib/src/components/NxToast/NxToast.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useRef, useContext, useEffect, useState } from 'react';
+import classnames from 'classnames';
+
+import { nxToastPropTypes, NxToastProps } from './types';
+import NxCloseButton from '../NxCloseButton/NxCloseButton';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+
+import ToastContext from './contexts';
+import { toastTypeMap } from './toastTypeMapping';
+
+const NxToast = (props: NxToastProps) => {
+  const { toastId, className, message, type, toastContainerRef, ...otherProps } = props,
+      toastClass = toastTypeMap[type].class,
+      toastIcon = toastTypeMap[type].icon,
+      toastIconLabel = toastTypeMap[type].iconLabel,
+      [animate, setAnimate] = useState(false),
+      classes = classnames('nx-toast', 'animate', className, toastClass,
+          {'slide-in': animate},
+          {'slide-out': !animate}
+      ),
+      toastContext = useContext(ToastContext),
+      closeBtnRef = useRef<HTMLButtonElement>(null);
+
+  //Fire effect to focus on the close button of the toast that is rendered
+  useEffect(() => {
+    const closeBtn = closeBtnRef.current;
+    if (closeBtn) {
+      closeBtn.focus();
+    }
+
+    //When component is mounted, add class "animate slide-in" to trigger animation
+    setAnimate(true);
+  }, []);
+
+  //Effect to shift focus to the next toast's close button
+  //Dependency on the global toasts array
+  useEffect(() => {
+    if (toastContainerRef) {
+      //Gets the first close button of the child from the parent toast container div
+      const closeBtn = toastContainerRef.current?.querySelectorAll('.nx-btn--close')[0] as HTMLButtonElement;
+      closeBtn.focus();
+    }
+  }, [toastContext?.toasts]);
+
+  const handleCloseClick = () => {
+    setAnimate(false);
+    //Wait for slide-out animation before removing toast from DOM
+    setTimeout(() => {
+      toastContext?.removeToast(toastId);
+    }, 250);
```

Comment:
I wonder if we should check for prefer reduce motion in JS and set this to 0 or something like that if it is set:
`const mediaQuery = window.matchMedia("(prefers-reduced-motion: reduce)");`

Reply:
> I wonder if we should check for prefer reduce motion in JS and set this to 0 or something like that if it is set: `const mediaQuery = window.matchMedia("(prefers-reduced-motion: reduce)");`

Is this still required? It seems that on Mac, the CSS is enough to handle it. Not sure what the behavior is on other OS .

Reply:
> Is this still required? It seems that on Mac, the CSS is enough to handle it. Not sure what the behavior is on other OS .

I think this suggestion only made sense if we didn't keep the fade.  Without either animation, but also without the change suggested here, I assume what would happen is you would click the X and then nothing would happen for a quarter-second and then only after that the toast would disappear.  The point, I believe, was just that the wait would be weird and unnecessary if there was no animation at all.

Reply:
> > Is this still required? It seems that on Mac, the CSS is enough to handle it. Not sure what the behavior is on other OS .
> 
> I think this suggestion only made sense if we didn't keep the fade. Without either animation, but also without the change suggested here, I assume what would happen is you would click the X and then nothing would happen for a quarter-second and then only after that the toast would disappear. The point, I believe, was just that the wait would be weird and unnecessary if there was no animation at all.

Turns out the CSS based approach is adequate in this context. The JS code is only required if we're using certain SVG, Canvas, or other complex JS-based animations. Since our animation aspect is strictly confined to using CSS transitions, it looks like we don't need to add the above JS code.

More on the topic [here.](https://www.joshwcomeau.com/react/prefers-reduced-motion/)

Reply:
> I think it still could've been necessary, but in our case isn't. Consider how it would have behaved if we had never chosen to add the fade in addition to the slide: when prefers-reduced-motion: reduce is in effect, no animation at all would happen on close, but the wait implemented here for the sake of animation still would happen. So it would just seem sluggish. But in reality that's moot because even in the prefers-reduced-motion: reduce case we still have one of the two animations (the fade) active and so the wait is still necessary.

From what I've heard we're removing the fade.  In that case the issue discussed in this thread becomes relevant once again.

Reply:
> From what I've heard we're removing the fade. In that case the issue discussed in this thread becomes relevant once again.

Though now that we've done away with the timer in favor of a `transitionend` listener, the way that the issue manifests is probably different.  With our current code, does `transitionend` fire at all when `prefers-reduced-motion: reduce` is in effect?

Reply:
>> I think it still could've been necessary, but in our case isn't. Consider how it would have behaved if we had never chosen to add the fade in addition to the slide: when `prefers-reduced-motion: reduce` is in effect, no animation at all would happen on close, but the _wait_ implemented here for the sake of animation still would happen. So it would just seem sluggish. But in reality that's moot because even in the `prefers-reduced-motion: reduce` case we still have one of the two animations (the fade) active and so the wait is still necessary.
>
>From what I've heard we're removing the fade. In that case the issue discussed in this thread becomes relevant once again.

Sorry for the late reply on this one, but I was waiting for the animation to be completed before dealing with `prefers-reduced-motion`. I have submitted the final animation to Jason, along with the reduced motion option which just includes a fade in / out and no sliding motion. Implemented here: 
[bf15401](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/bf154012fdb55ccc7fa5c5d4edea02e11c2f58aa)